packages/block-library/src/post-content/edit.js

```diff
@@ -13,19 +12,11 @@ import PostContentInnerBlocks from './inner-blocks';
 export default function PostContentEdit( {
 	context: { postId: contextPostId, postType: contextPostType },
 } ) {
-	const { id: currentPostId, type: currentPostType } = useSelect(
-		( select ) => select( 'core/editor' ).getCurrentPost() ?? {}
-	);
```

Comment:
This reverts #24010 

It seems the checks were not good enough IMO. It relies on something that doesn't give us semantic information about the context but it was more an implementation detail.

Reply:
Yeah, I agree that these original checks were not great, but there simply wasn't another way to do it with the information available at the time. There didn't seem to be a super clear way to add the necessary information to the context either... The problem is really, given a global context with a postID, how do you prevent nested context consumers from rendering recursively infinitely?

Reply:
It seems we need a context value to indicate the post content id we're currently rendering (and not just the "post id").

Reply:
It's not fixed entirely, but the important breakage in the template editor is fixed by 46724a8